cmflib/commands/metadata/export.py

```diff
@@ -18,60 +18,108 @@
-        if not os.path.exists(mlmd_file_name):
-            return f"ERROR: {mlmd_file_name} doesn't exists in the {current_directory}."
-
-
-        # setting directory where mlmd file will be dumped
-        if self.args.json_file_name:
-            if not os.path.isdir(self.args.json_file_name):
-                temp = os.path.dirname(self.args.json_file_name)
-                if temp != "":
-                    current_directory = temp
-                if os.path.exists(current_directory):
-                    full_path_to_dump  = self.args.json_file_name
-                else:
-                    return f"{current_directory} doesn't exists."
-            else:
-                return "Provide path with file name."
+        if not self.args.file_name:         # If self.args.file_name is None or an empty list ([]). 
+            mlmd_file_name = "./mlmd"       # Default path for mlmd file name.
+        elif len(self.args.file_name) > 1:  # If the user provided more than one file name.   
+            return "Error: You can only provide one file name using the -f flag."
+        elif not self.args.file_name[0]:    # self.args.file_name[0] is an empty string ("").
+            return "Error: Missing File name"
         else:
-            full_path_to_dump = os.getcwd() + f"/{self.args.pipeline_name}.json"
-
-        # initialising cmfquery class
+            mlmd_file_name = self.args.file_name[0].strip() # Removing starting and ending whitespaces.
+            if mlmd_file_name == "mlmd":
+                mlmd_file_name = "./mlmd"
+        
+        current_directory = os.path.dirname(mlmd_file_name)
+        if not os.path.exists(mlmd_file_name): 
+            return f"Error: {mlmd_file_name} doesn't exists in {current_directory} directory."
+
+        # Initialising cmfquery class.
         query = cmfquery.CmfQuery(mlmd_file_name)
 
-        # check if pipeline exists in mlmd 
-        pipeline = query.get_pipeline_id(self.args.pipeline_name)
+        # Check if pipeline exists in mlmd .
+        if self.args.pipeline_name is not None and len(self.args.pipeline_name) > 1:   
+            return "Error: You can only provide one pipeline name using the -p flag."
+        elif not self.args.pipeline_name[0]:    # self.args.pipeline_name[0] is an empty string (""). 
+            return "Error: Missing pipeline name"
+        else:
+            pipeline_name = self.args.pipeline_name[0]
+        
+        pipeline = query.get_pipeline_id(pipeline_name)
 
         if pipeline > 0:
-            # pulling data from local mlmd file
-            json_payload = query.dumptojson(self.args.pipeline_name,None)
-
-            # write metadata into json file
+            if not self.args.json_file_name:         # If self.args.json_file_name is None or an empty list ([]). 
+                json_file_name = self.args.json_file_name
+            elif len(self.args.json_file_name) > 1:  # If the user provided more than one json file name. 
```

Comment:
This comment is a good example of what the number is exactly referring to. Can you please update other files to reflect something similar where the intent is clearly known and it is also known that what will be the impact if a value is changed.

Reply:
Sure.